flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/channel/ChannelStateWriteRequestExecutorFactory.java

```diff
@@ -56,15 +64,32 @@ ChannelStateWriteRequestExecutor getOrCreateExecutor(
             JobVertexID jobVertexID,
             int subtaskIndex,
             CheckpointStorage checkpointStorage,
+            CheckpointStorageAccess checkpointStorageAccess,
             int maxSubtasksPerChannelStateFile,
-            boolean startExecutor) {
+            boolean startExecutor,
+            boolean unifyFileMerging) {
         synchronized (lock) {
+            int actualMaxSubtasksPerChannelStateFile = maxSubtasksPerChannelStateFile;
+            ChannelStateWriteRequestDispatcher dispatcher = null;
+            if (!unifyFileMerging) {
+                dispatcher =
+                        new ChannelStateWriteRequestDispatcherImpl(
+                                checkpointStorage, jobID, new ChannelStateSerializerImpl());
+            } else {
+                dispatcher =
+                        new UnifyMergingChannelStateWriteRequestDispatcher(
```

Comment:
IIUC, all subtasks of one job share same dispatcher means all subtasks will write same channel state file, right?

If so, all subtasks will share single thread, I'm afraid the performance isn't enough.

Reply:
Right, all subtasks will share single thread in this PR, good catch!

I plan to change it to a subtask using an executor, like things before FLINK-26803. Is this feasible?